bluebrain/deployment/environments/applications_hpc.yaml

```diff
@@ -41,13 +41,13 @@ spack:
     # - neuron%nvhpc+tests+gpu+openmp+coreneuron
     # - neuron%nvhpc+tests+gpu+openmp+nmodl+sympy+coreneuron
     - nest@2.20.1
-    - neurodamus-core~common
-    - neurodamus-core+common
+    - neurodamus-core~common~~coreneuron
+    - neurodamus-core+common~~coreneuron
```

Comment:
I think `neurodamus-core` could be removed. We now only depend on it via the submodule in `py-neurodamus`

Reply:
I see this still:
```
bluebrain/repo-bluebrain/packages/neurodamus-model/package.py
31:    depends_on("neurodamus-core@" + core_v, type="build", when="@" + this_version)
```
Should we then remove the old stuff that uses `version_from_model_core_dep`, too?

Reply:
If we can clean up the older versions, I think that would be a good start to simplify our package structure.

Reply:
I think everything related to `neurodamus-core` could be removed. We now only depend to it by the submodule of `py-neurodamus`.
I think also we need to remove all the `py-neurodamus` versions prior to `2.9.0` that expect that there is a `neurodamus-core` package and doesn't bring the submodule as well